bundles/org.openhab.binding.lghombot/src/main/java/org/openhab/binding/lghombot/internal/LGHomBotHandler.java

```diff
@@ -0,0 +1,660 @@
+                        break;
+                    case "THURSDAY":
+                        thursday = state;
+                        break;
+                    case "FRIDAY":
+                        friday = state;
+                        break;
+                    case "SATURDAY":
+                        saturday = state;
+                        break;
+                    case "SUNDAY":
+                        sunday = state;
+                        break;
+                    default:
+                        break;
+                }
+
+            }
+            if (!currentMonday.equals(monday)) {
+                currentMonday = monday;
+                updateState(CHANNEL_MONDAY, StringType.valueOf(monday));
+            }
+            if (!currentTuesday.equals(tuesday)) {
+                currentTuesday = tuesday;
+                updateState(CHANNEL_TUESDAY, StringType.valueOf(tuesday));
+            }
+            if (!currentWednesday.equals(wednesday)) {
+                currentWednesday = wednesday;
+                updateState(CHANNEL_WEDNESDAY, StringType.valueOf(wednesday));
+            }
+            if (!currentThursday.equals(thursday)) {
+                currentThursday = thursday;
+                updateState(CHANNEL_THURSDAY, StringType.valueOf(thursday));
+            }
+            if (!currentFriday.equals(friday)) {
+                currentFriday = friday;
+                updateState(CHANNEL_FRIDAY, StringType.valueOf(friday));
+            }
+            if (!currentSaturday.equals(saturday)) {
+                currentSaturday = saturday;
+                updateState(CHANNEL_SATURDAY, StringType.valueOf(saturday));
+            }
+            if (!currentSunday.equals(sunday)) {
+                currentSunday = sunday;
+                updateState(CHANNEL_SUNDAY, StringType.valueOf(sunday));
+            }
+        }
+    }
+
+    private void parseImage() {
+        if (!isLinked(CHANNEL_CAMERA)) {
+            return;
+        }
+        final int size = 320 * 240;
+        String url = buildHttpAddress("/images/snapshot.yuv");
+        byte[] yuvData = HttpUtil.downloadData(url, null, false, size * 2).getBytes();
+
+        currentImage = CameraUtil.parseImageFromBytes(yuvData);
+    }
+
+    /** Parse the maps.html file to find the black-box filename. */
+    private String findBlackBoxFile() {
+        String url = buildHttpAddress("/sites/maps.html");
+        try {
+            String htmlString = HttpUtil.executeUrl("GET", url, 1000);
+            int idx = htmlString.indexOf("blkfiles");
+            return "/.../usr/data/blackbox/" + htmlString.substring(idx + 13, idx + 50);
+        } catch (IOException e1) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e1.getMessage());
+        }
+        return "";
+    }
+
+    private void parseMap() {
+        if (!isLinked(CHANNEL_MAP)) {
+            return;
+        }
+        final int tileSize = 10;
```

Comment:
These 4 could be real constants

Reply:
Real constants, as in "static final"? Or class level constants? What is the benefit of class level constants if they are only relevant to one method?

Reply:
Also I couldn't reply to the question about the map(a -> a). I'm not sure what the amp(a ->a) does, I looked up an example of how to get ahold of a local IP4 address. What I want to do is find the IP or broadcast-address configured in the OpenHAB networks settings.

Reply:
It looks like the discovery still works without it so I removed it.

Reply:
`map` can be used to get transform data into other data. But `a -> a` means just return the same value as was passed. Hence it doesn't look useful and makes sense it still works when you remove it.

If you push that change and you're give the ok, we can merge this binding.

Reply:
> Real constants, as in "static final"? Or class level constants? What is the benefit of class level constants if they are only relevant to one method?

Fair question, answer is that that is how we mostly do it as Java developers. Normally a class is only about a certain subjectso the constant would be highly related to all other methods in the class.